app/src/main/java/edu/agh/susgame/front/ui/component/game/map/GameGraphComponent.kt

```diff
@@ -0,0 +1,148 @@
+package edu.agh.susgame.front.ui.component.game.map
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.detectTransformGestures
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clipToBounds
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import edu.agh.susgame.front.model.graph.GameGraph
+import edu.agh.susgame.front.model.graph.Node
+import edu.agh.susgame.front.ui.theme.PaddingM
+import edu.agh.susgame.front.ui.util.ZoomState
+
+private const val buttonWidth = 70
+private const val buttonHeight = 30
+
+@Composable
+internal fun GameGraphComponent(mapState: GameGraph) {
+    val changeNodeInfoVisibility = remember { mutableStateOf(false) }
+    val nodeToShow = remember { mutableIntStateOf(0) }
```

Comment:
Ja bym to zwinął do jednej zmiennej np. w taki sposób:
(`crtl + a` -> `Apply patch from clipboard...`)

```
Index: app/src/main/java/edu/agh/susgame/front/ui/component/game/map/GameGraphComponent.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/edu/agh/susgame/front/ui/component/game/map/GameGraphComponent.kt b/app/src/main/java/edu/agh/susgame/front/ui/component/game/map/GameGraphComponent.kt
--- a/app/src/main/java/edu/agh/susgame/front/ui/component/game/map/GameGraphComponent.kt	(revision bedd3780ee4d695f57338231f48ad1b7fa85a13d)
+++ b/app/src/main/java/edu/agh/susgame/front/ui/component/game/map/GameGraphComponent.kt	(date 1717663016478)
@@ -16,10 +16,10 @@
 import androidx.compose.material3.Button
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.MutableState
-import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clipToBounds
@@ -39,8 +39,7 @@
 
 @Composable
 internal fun GameGraphComponent(mapState: GameGraph) {
-    val changeNodeInfoVisibility = remember { mutableStateOf(false) }
-    val nodeToShow = remember { mutableIntStateOf(0) }
+    var inspectedNodeId by remember { mutableStateOf<Int?>(null) }
     val zoomState = remember {
         ZoomState(
             maxZoomIn = 2f,
@@ -107,8 +106,7 @@
                             height = buttonHeight.dp
                         ),
                     onClick = {
-                        changeNodeInfoVisibility.value = true
-                        nodeToShow.intValue = node.id
+                        inspectedNodeId = node.id
                     },
                 ) {
                     Text(
@@ -118,31 +116,37 @@
                 }
             }
         }
-        if (changeNodeInfoVisibility.value) {
-            mapState.nodes[nodeToShow.intValue]?.let { ShowInfo(node = it,changeNodeInfoVisibility) }
+        inspectedNodeId?.let { nodeId ->
+            mapState.nodes[nodeId]?.let {
+                ShowInfo(
+                    node = it,
+                    onExit = {
+                        inspectedNodeId = null
+                    },
+                )
+            }
         }
     }
 }
 
 
 @Composable
-private fun ShowInfo(node: Node, showBox: MutableState<Boolean>) {
+private fun ShowInfo(node: Node, onExit: () -> Unit) {
     Box(
         modifier = Modifier
             .background(Color.Cyan)
             .padding(PaddingM),
-        ) {
-        Row() {
+    ) {
+        Row {
             Column(
                 verticalArrangement = Arrangement.Center,
                 horizontalAlignment = Alignment.CenterHorizontally
             ) {
                 Text(node.getInfo())
             }
-            Button(onClick = { showBox.value = false }) {
+            Button(onClick = { onExit() }) {
                 Text("X")
             }
         }
     }
 }
-
```

Reply:
dodałem